registry/client/src/settings/Edit.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+    Edit,
+    SimpleForm,
+    TextInput,
+    SelectInput,
+    BooleanInput,
+} from 'react-admin';
+
+import * as validators from '../validators';
+import {choices, keys} from './dataTransform';
+
+const Title = ({record}) => {
+    return (<span>{record ? `Setting "${record.key}"` : ''}</span>);
+};
+
+const InputForm = (props) => {
+    switch(props.record.key) {
```

Comment:
I think we shouldn't hardcode any keys here. Addition of the new setting should be a matter of adding new row to DB and maybe adding new key to Joi schema on server side. So 1, max 2 update points.

Reply:
Think about the logic when server returns also: key type & possible values for select.